homeassistant/components/lektrico/sensor.py

```diff
@@ -0,0 +1,296 @@
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_FRIENDLY_NAME,
+    PERCENTAGE,
+    UnitOfElectricCurrent,
+    UnitOfElectricPotential,
+    UnitOfEnergy,
+    UnitOfPower,
+    UnitOfTemperature,
+    UnitOfTime,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import LektricoDeviceDataUpdateCoordinator
+from .const import DOMAIN
+from .entity import LektricoEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class LektricoSensorEntityDescription(SensorEntityDescription):
+    """A class that describes the Lektrico sensor entities."""
+
+    value_fn: Callable[[Any], float | str | int]
+
+
+SENSORS_FOR_CHARGERS: tuple[LektricoSensorEntityDescription, ...] = (
+    LektricoSensorEntityDescription(
+        key="state",
+        translation_key="state",
+        value_fn=lambda data: str(data.charger_state),
+    ),
+    LektricoSensorEntityDescription(
+        key="charging_time",
+        translation_key="charging_time",
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        value_fn=lambda data: int(data.charging_time),
+    ),
+    LektricoSensorEntityDescription(
+        key="power",
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        value_fn=lambda data: float(data.instant_power) / 1000,
+    ),
+    LektricoSensorEntityDescription(
+        key="energy",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        value_fn=lambda data: float(data.session_energy) / 1000,
+    ),
+    LektricoSensorEntityDescription(
+        key="temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        value_fn=lambda data: float(data.temperature),
+    ),
+    LektricoSensorEntityDescription(
+        key="lifetime_energy",
+        translation_key="lifetime_energy",
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        value_fn=lambda data: int(data.total_charged_energy),
+    ),
+    LektricoSensorEntityDescription(
+        key="installation_current",
+        translation_key="installation_current",
+        device_class=SensorDeviceClass.CURRENT,
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        value_fn=lambda data: int(data.install_current),
+    ),
+    LektricoSensorEntityDescription(
+        key="limit_reason",
+        translation_key="limit_reason",
+        value_fn=lambda data: str(data.current_limit_reason),
+    ),
```

Comment:
What can the state be?

Reply:
The limit_reason can be:
"No limit", "Installation current", 
"User limit", "Dynamic limit", "Schedule",
"EM offline", "EM", "OCPP"

Reply:
Idem, it should be an enum sensor

Reply:
I made the changes, thanks.